core/src/main/java/com/spotify/ffwd/output/CoreOutputManager.java

```diff
@@ -181,6 +194,27 @@ public final Long getHLLPRefreshPeriodLimit() {
         } else {
             this.cardinalityLimit = null;
         }
+
+        if (!Strings.isNullOrEmpty(dynamicTagsFile)) {
+            final File file = new File(dynamicTagsFile);
+            if (file.exists() && file.canRead()) {
+                tagRefreshingThread = Executors.newSingleThreadScheduledExecutor(
+                    new ThreadFactoryBuilder().setNameFormat("tag-refreshing-thread").build());
```

Comment:
Is the used of an executor here justified ? It inside a constructor the tag reference is not available to any thread yet.

Reply:
Oh, can you explain why the tag reference isn't available? It seemed like it was? What did you have in mind instead?